kernelci/cli/docker.py

```diff
@@ -170,9 +167,15 @@ def __call__(self, configs, args):
         buildargs = dict(
             barg.split('=') for barg in args.build_arg
         ) if args.build_arg else {}
-        _, build_log = self._build_image(
-            dockerfile, name, buildargs, args.no_cache
-        )
+        try:
+            _, build_log = self._build_image(
+                dockerfile, name, buildargs, args.no_cache
+            )
+        except docker.errors.BuildError as exc:
+            if args.verbose:
+                self._dump_log(exc.build_log)
```

Comment:
The issue here is that the log is now only printed when the exception is caught.  Instead, the `build_log` variable should be populated here and dumped in one place e.g. a `finally` block.

Reply:
The `finally` block is run regardless of whether there was an exception or not, so no need to duplicate the code (well there should never be a need for duplicating code anyway to keep it [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)).

Reply:
Done, fixed. 
`build_log` exist only if exception didnt happen so i used else:

Reply:
Erm, that's not an ideal implementation but I think I'll merge and send a change with what I mean to avoid playing nuclearcat and mouse ;)

Reply:
Actually pylint hinted me it is not necessary, because we have return in case of exception, so we will not reach printing build_log anyway